### PR TITLE
Update CONTRIBUTING and HACKING docs

### PR DESCRIPTION
**Pull request by kargakis**
@danmcp @stevekuznetsov PTAL

### CODE REVIEW

File: CONTRIBUTING.adoc [hunk 1]
```diff
@@ -368,7 +368,8 @@ Right now you can see what's happening with OpenShift development at:
 
 https://github.com/openshift/origin[github.com/openshift/origin]
 
-Ready to play with some code? Hop down and read up on our link:#_the_roadmap[roadmap] for ideas on where you can contribute.
+Ready to play with some code? Hop down and read up on our link:#the-roadmap[roadmap] for ideas on where you can contribute.
+You can also try to take a stab in any issue tagged with the https://github.com/openshift/origin/issues?q=is%3Aissue+is%3Aopen+label%3Ahelp-wanted[help-wanted] label.
```
Comment (danmcp) [inline CONTRIBUTING.adoc]: Should be 'take a stab at'

File: HACKING.md [hunk 2]
```diff
@@ -203,6 +203,11 @@ Run the extended tests with:
 
     $ test/extended/core.sh
 
+This suite comprises many smaller suites, which are found under `test/extended` and can be run individually by
+specifying them using `--ginkgo.focus` and a regex filter:
```
Comment (danmcp) [inline HACKING.md]: Would remove "them using"

File: CONTRIBUTING.adoc (hunk 1 above)
Comment (kargakis) [inline CONTRIBUTING.adoc]: ok

File: HACKING.md (hunk 2 above)
Comment (kargakis) [inline HACKING.md]: ok

File: HACKING.md [hunk 3]
```diff
@@ -203,6 +203,11 @@ Run the extended tests with:
 
     $ test/extended/core.sh
 
+This suite comprises many smaller suites, which are found under `test/extended` and can be run individually by
+specifying `--ginkgo.focus` and a regex filter:
+
+    $ test/extended/core.sh --ginkgo.focus=<regex>
```
Comment (stevekuznetsov) [inline HACKING.md]: Do we link to `test/extended/README` here?

**Comment by stevekuznetsov** [comment]
changes seem fine

File: HACKING.md (hunk 3 above)
Comment (kargakis) [inline HACKING.md]: Good idea, I wasn't aware of its existense. Updated

**Comment by kargakis** [comment]
comments addressed, [merge]

**Comment by kargakis** [comment]
@danmcp  should we run doc pulls through CI?

**Comment by danmcp** [comment]
@kargakis Yeah, that's the lesser of the evils.

**Pull request merged**